### DISCUSSION TITLE
Auto budget not adding new values at the end of period

### DISCUSSION THREAD

**Discussion by goyban** [firefly-iii/firefly-iii#7809] [General]
I installed Firefly via Docker-compose
Everything is working correctly but the auto budget, itstays at 0 on the new period.

I saw firefly_iii_cron container running and I assumed its working, but I see in the logs that there is a problem:

`crond: USER root pid  14 cmd wget -qO- http://app:8080/api/v1/cron/REPLACEME

wget: server returned error: HTTP/1.1 404 Not Found`

1. Is this issue "auto budget not being add" related to this?
2. If so how what is the problem and what should I change? The port set on docker-compose file is: '8564:8080'

Please let me know if I need to explain more or share more info.

**Comment by JC5** [comment]
Yes, that is the problem. It seems that URL can't be reached by your docker host.

What happens when you try it with this URL? `http://127.0.0.1:8564/api/v1/cron/REPLACEME`

I assume that you did in fact replace "REPLACEME" with the actual API code?

**Reply by goyban** [reply]
Thanks for the quick reply, so it turns out I forgot to add the API on .env file as well. 
Sorry for that, it was a user error I thought something is wrong with the ports because of the error.
Now the log is like this:
`crond: crond (busybox 1.36.1) started, log level 8`
It is working now, right?

How should I correct the last budget because I missed last Monday? Also last week's amount hasn't been added. (Should I add them manually for this week or there is a way to make it work?)
![image](https://github.com/firefly-iii/firefly-iii/assets/23245225/f2dc15e8-c957-4e5a-9bd4-455efdbe1421)

**Reply by JC5** [reply]
1. Make sure you use the `app:8080` url, not my suggestion, it probably won't work.
2. To force a cron job for a specific date, use `docker exec -it [firefly-iii-container] php artisan firefly:cron --force --date=2023-08-01`

**Reply by goyban** [reply]
Perfect!
Thanks, it worked, also thanks for your good work.

**Reply by Zyadon7** [reply]
Can the --force and --date parameters be translated into the web request cron command?  
Example $ wget -qO- http://123.123.123.123:8080/api/v1/cron/SuperSecretAPIkey
